src/core/libmaven/mzMassSlicer.cpp

```diff
@@ -366,120 +366,14 @@ void MassSlices::_reduceSlices()
                 secondSlice->ionCount = -1.0f;
             }
         }
-        sendSignal("Reducing slices…", first - begin(slices), slices.size());
+        sendSignal("Reducing redundant slices…",
```

Comment:
I would suggest moving the code for deleting slices here if possible. Right now if I go through algorithmB, it looks like there is one call for reducing slices, then another loop for removing slices, then a call for removing duplicate slices and so on. Maybe we can also merge `reduceSlices` and `removeDuplicateSlices` while we're at it? I've not seen the code for both methods in detail so I'll let you take that call.

Reply:
or just call `removeDuplicateSlices` from within `reduceSlices`.

Reply:
Right. The deletion part definitely belongs here.

But just like you, I am not familiar with the `removeDuplicateSlices` method, and I suspect its largely useless. We had agreed to get rid of it some time ago. I will do that now.

Reply:
I remember we had a conversation about the utility of `removeDuplicateSlices` while testing, but not whether it would make a difference in the output. I too suspect it shouldn't affect the output but worth a sanity check, no? Do we have saved output from a previous version of this branch?

Reply:
Not at the moment. But we did check while deciding. This method would, at best, lead to reduction of 1 or 2 slices among several thousands.

Reply:
Cool, then. Resolving.